### PR TITLE
G11

### PR DESCRIPTION
**Pull request by toasterpm87**
# Description

Kraut space magic

---

# Tasks

- [x] Task
- [x] Completed Task

---

<details><summary><h1>Media</h1></summary>
<p>

image

</p>
</details>

---

# Changelog

:cl:
- add: Added fun!
- tweak: Tweaked fun!
- fix: Fixed fun!
- remove: Removed fun!

### CODE REVIEW

**Comment by toasterpm87** [comment]
image

**Comment by toasterpm87** [comment]
I had a revelation in the shower, i should add firerate, to decrease the between burst time

**Comment by toasterpm87** [comment]
I will also add it to deltav

**Comment by DEATHB4DEFEAT** [comment]
firefox_T55jDM3kPH

**Comment by toasterpm87** [comment]
> firefox_T55jDM3kPH
> 

Its already fixed

**Comment by DEATHB4DEFEAT** [comment]
> > firefox_T55jDM3kPH
> 
> Its already fixed

Push the fix then, current version is supposedly invalid

**Comment by toasterpm87** [comment]
i fixed it

**Comment by toasterpm87** [comment]
let me test something first then we can merge it

**Comment by toasterpm87** [comment]
g11magin
I am done

**Comment by toasterpm87** [comment]
its ready

File: Resources/Prototypes/SimpleStation14/Entites/Weapons/rifle.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+- type: entity
+  name: G11
+  parent: BaseWeaponRifle
+  id: WeaponRifleG11
+  description: A unique rifle with an clock-like internals and a top fed magazine. Uses .20 rifle ammo. #switch to caseless when you have the time
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Entites/Weapons/rifle.yml:5]:
```suggestion
  description: A unique rifle with a clock-like internals and a top fed magazine. Uses .20 rifle ammo. # TODO Use caseless ammunition
```

File: Resources/Prototypes/SimpleStation14/Entites/Weapons/rifle.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            - CartridgeCaselessRifle
+  - type: ContainerContainer
+    containers:
+      gun_magazine: !type:ContainerSlot
+      gun_chamber: !type:ContainerSlot
+  - type: MagazineVisuals
+    magState: mag
+    steps: 1
+    zeroVisible: true
+  - type: Appearance
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Entites/Weapons/rifle.yml:50]:
```suggestion
    - type: Sprite
      sprite: SimpleStation14\weapons\G11.rsi
      layers:
        - state: base
          map: ["enum.GunVisualLayers.Base"]
        - state: mag-0
          map: ["enum.GunVisualLayers.Mag"]
    - type: Clothing
      sprite: SimpleStation14\weapons\G11.rsi
    - type: Gun
      fireRate: 10
      soundGunshot:
        path: /Audio/Weapons/Guns/Gunshots/ltrifle.ogg
      selectedMode: Burst
      availableModes:
      - Burst
      - FullAuto
    - type: ItemSlots
      slots:
        gun_magazine:
          name: Magazine
          startingItem: MagazineCaselessRifle
          insertSound: /Audio/Weapons/Guns/MagIn/ltrifle_magin.ogg
          ejectSound: /Audio/Weapons/Guns/MagOut/ltrifle_magout.ogg
          priority: 2
          whitelist:
            tags:
              - MagazineCaselessRifle
        gun_chamber:
          name: Chamber
          startingItem: CartridgeCaselessRifle
          priority: 1
          whitelist:
            tags:
              - CartridgeCaselessRifle
    - type: ContainerContainer
      containers:
        gun_magazine: !type:ContainerSlot
        gun_chamber: !type:ContainerSlot
    - type: MagazineVisuals
      magState: mag
      steps: 1
      zeroVisible: true
    - type: Appearance
```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]
Someone else should look at the sprites

File: Resources/Prototypes/SimpleStation14/Entites/Weapons/rifle.yml (hunk 1 above)
Comment (toasterpm87) [reply Resources/Prototypes/SimpleStation14/Entites/Weapons/rifle.yml:5]: this needs to be removed entirely since it is already caseless

**Comment by LovelyLophi** [comment]
I personally feel that the gun visuals do not match the current visual style of the game and this should not be merged until it does.

**Comment by DEATHB4DEFEAT** [comment]
There are also tests failing

File: Resources/Prototypes/SimpleStation14/Entites/Weapons/rifle.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  components:
+    - type: Sprite
+      sprite: SimpleStation14\weapons\G11.rsi
+      layers:
+        - state: base
+          map: ["enum.GunVisualLayers.Base"]
+        - state: mag-0
+          map: ["enum.GunVisualLayers.Mag"]
+    - type: Clothing
+      sprite: SimpleStation14\weapons\G11.rsi
```
Comment (VMSolidus) [inline Resources/Prototypes/SimpleStation14/Entites/Weapons/rifle.yml:15]:
```suggestion
      sprite: SimpleStation14/weapons/G11.rsi
      layers:
        - state: base
          map: ["enum.GunVisualLayers.Base"]
        - state: mag-0
          map: ["enum.GunVisualLayers.Mag"]
    - type: Clothing
      sprite: SimpleStation14/weapons/G11.rsi
```

**Review by VMSolidus — CHANGES_REQUESTED** [review]
Guh, backward slashes.

**Pull request closed without merging**